### ISSUE TITLE
Add support for security setups to logbook-servlet

### ISSUE THREAD

**Issue by whiskeysierra** [zalando/logbook#17]
In our common setup we would like to have a filter chain setup like this:

1. logging, 1st pass
2. security
2. logging, 2nd pass

The first logging filter should log both, request and response, if and only if the response is a 401 (or 403?) and it should ignore the request body. This is desired because unauthorized requests can potentially be DoS attacks with very large requests to take our service down.

The second logging filter logs requests as they come in and responses as they go out - business as usual. The only important thing is, that any request/response is only **logged exactly once**.

**Comment by lukasniemeier-zalando** [comment]
- consuming the request / response should be configurable per filter instance (i.e. body is optional)
- _optional_ communication channel between filter instances to have "exactly once" semantic when logging request or response

**Comment by whiskeysierra** [comment]
Just for clarification: The proposed setup will lead to log files that are not 100% correct in terms of chronological order. Any request to an external service inside the security filter will be logged before the original request (e.g. to verify OAuth token):

    OAuth request
    OAuth response
    Original request
    Original response

while the real order (as it actually happened) is more like:

    Original request
    OAuth request
    OAuth response
    Original response

I don't think this is an issue and it will even be consistent between authenticated/authorized requests and unauthenticated/unauthorized requests.

Basically: *Every request made before the actual logging will be done out of order.*
